test/integration/api/forms/forms.js

```diff
@@ -772,6 +772,25 @@ describe('api: /projects/:id/forms (create, read, update)', () => {
               text.should.equal(testData.forms.simple);
             }))));
 
+      it('should return standard body if incorrect ETag provided', testService((service) =>
+        service.login('alice', (asAlice) =>
+          asAlice.get('/v1/projects/1/forms/simple.xml')
+            .set('If-None-Match', 'a-random-string')
+            .expect(200)
+            .then(({ text }) => {
+              text.should.equal(testData.forms.simple);
+            }))));
+
+      it('should return 304 if correct ETag provided', testService((service) =>
+        service.login('alice', async (asAlice) => {
+          const res = await asAlice.get('/v1/projects/1/forms/simple.xml').expect(200);
```

Comment:
Form XML isn't stored as a blob, so I don't think this test is exercising the new functionality. We do store XLSForms as blobs though.